astropy/cosmology/funcs/tests/test_funcs.py

```diff
@@ -299,15 +308,23 @@ def test_z_at_value_bracketed(method):
             rtol=1e-6,
         )
 
-    with pytest.raises(core.CosmologyError):
-        with pytest.warns(AstropyUserWarning, match=r"fval is not bracketed"):
-            z_at_value(
-                cosmo.angular_diameter_distance,
-                1500 * u.Mpc,
-                method=method,
-                bracket=(3.9, 5.0),
-                zmin=4.0,
-            )
+    if not PYTEST_LT_8_0 and method == "Bounded":
+        ctx = pytest.warns(
+            UserWarning, match="Option 'bracket' is ignored by method Bounded"
+        )
+    else:
+        ctx = nullcontext()
+
+    with pytest.raises(core.CosmologyError), pytest.warns(
+        AstropyUserWarning, match=r"fval is not bracketed"
```

Comment:
The `ctx` from line 204 can be used here, if given a distinct name.

Reply:
I don't think so. See https://github.com/astropy/astropy/pull/15054#discussion_r1261659206